pants

```diff
@@ -72,42 +72,46 @@ EOF
 }
 
 # The high-level flow:
-# 1.) Resolve the Python interpreter, first reading from the env var $PYTHON,
+# 1.) Resolve the Pants version from pants.ini or from the latest stable
+#     release to PyPi.
+# 2.) Resolve the Python interpreter, first reading from the env var $PYTHON,
 #     then reading from pants.ini, then determining the default based off of
-#     which Python versions the current `pants_version` supports.
-# 2.) Grab pants version from pants.ini or default to latest.
+#     which Python versions the Pants version supports.
 # 3.) Check for a venv via a naming/path convention and execute if found.
 # 4.) Otherwise create venv and re-exec self.
 #
-# After that pants itself will handle making sure any requested plugins
+# After that, Pants itself will handle making sure any requested plugins
 # are installed and up to date.
 
-function determine_default_python_exe {
+function determine_pants_version {
   pants_version="$(get_pants_ini_config_value 'pants_version')"
-  # If `pants_version` not pinned in `pants.ini`, default to what the most
-  # recent stable release supports.
-  if [[ -z "${pants_version}" ]]; then
+  if [[ -n "${pants_version}" ]]; then
+    echo "${pants_version}"
+  else
+    curl -sSL https://pypi.python.org/pypi/pantsbuild.pants/json |
+      python -c "import json, sys; print(json.load(sys.stdin)['info']['version'])"
+  fi
+}
+
+function determine_default_python_exe {
+  pants_version="$1"
+  pants_major_version="$(echo "${pants_version}" | cut -d '.' -f1)"
+  pants_minor_version="$(echo "${pants_version}" | cut -d '.' -f2)"
```

Comment:
This could be extracted into a function, like:
```bash
function extract_version_field {
  version_string="$1"
  field_index="$2"
  echo "$version_string" | cut -d '.' "-f${field_index}"
}
```

It's not clear that's necessary right now and I believe others would probably argue against it right now, but I think for a well-defined operation like this it could help to add clarity.